src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -468,9 +468,16 @@ void CodegenNeuronCppVisitor::print_make_instance() const {
     const auto codegen_float_variables_size = codegen_float_variables.size();
     for (int i = 0; i < codegen_float_variables_size; ++i) {
         const auto& float_var = codegen_float_variables[i];
-        printer->fmt_line("&_ml.template fpfield<{}>(0){}",
-                          i,
-                          i < codegen_float_variables_size - 1 ? "," : "");
+        if (float_var->is_array()) {
```

Comment:
Does this work the same with `RANGE` and `ASSIGNED` variables?
For example:
```
NEURON {
    SUFFIX leonhard
    RANGE a
}

ASSIGNED {
    a[2]
}

STATE {
  x
  y[2]
}

INITIAL {
    x = 42.0
    y[0] = 0.1
    y[1] = -1.0
    a[0] = 2
    a[1] = 0.5
}

BREAKPOINT {
    SOLVE dX METHOD cnexp
}

DERIVATIVE dX {
    x' = (y[0] + y[1])*x*a[0]*a[1]
}
```

Reply:
Yes.